File: Binary Search/4. Find Minimum in Rotated Sorted Array/solution.exs
defmodule Solution do
  @spec find_min(nums :: [integer]) :: integer
  def find_min([h | tail]) do
    recursive(tail, h, h)
  end

  def recursive([], _prev, head), do: head

  def recursive([h | tail], prev, head) when prev < h, do: recursive(tail, h, head)

  def recursive([h | _], prev, _head), do: h
end

File: Binary Search/4. Find Minimum in Rotated Sorted Array/solution.scala
object Solution {
  def findMin(nums: Array[Int]): Int = {
    var l = 0
    var r = nums.length - 1
    while (l < r) {
      val mid = l + (r - l) / 2
      if (nums(mid) > nums(r))
        l = mid + 1
      else
        r = mid
    }
    nums(l)
  }
}
